instance/models/server.py

```diff
@@ -287,8 +289,8 @@ def public_ip(self):
 
         try:
             public_addr = openstack.get_server_public_address(self.os_server)
-        except novaclient.exceptions.NotFound:
-            return None  # This server has been deleted.
+        except Exception:  # pylint: disable=broad-except
```

Comment:
You should be bale to catch the requests exceptions all at once, if I remeber correctly they all depend on a base custom exception.

It might seem small at this point, but bugs caused/hidden by broad except can be hell to figure out - and it only gets worse as the code base expands.